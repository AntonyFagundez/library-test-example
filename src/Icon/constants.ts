export enum IconSize {
  XXSMALL = "XXSMALL",
  XSMALL = "XSMALL",
  MSMALL = "MSMALL",
  SMALL = "SMALL",
  REGULAR = "REGULAR",
  XREGULAR = "XREGULAR",
  MEDIUM = "MEDIUM",
  XMEDIUM = "XMEDIUM",
  LARGE = "LARGE",
  BIG = "BIG",
  XBIG = "XBIG",
  XXBIG = "XXBIG",
}
