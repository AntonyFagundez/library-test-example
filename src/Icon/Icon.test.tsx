import { render } from "@testing-library/react";

import { IconSize } from "./constants";
import { Icon } from ".";
import React from "react";

const ICON_SIZE = [
  IconSize.XXSMALL,
  IconSize.XSMALL,
  IconSize.MSMALL,
  IconSize.SMALL,
  IconSize.REGULAR,
  IconSize.XREGULAR,
  IconSize.MEDIUM,
  IconSize.XMEDIUM,
  IconSize.LARGE,
  IconSize.BIG,
  IconSize.XBIG,
  IconSize.XXBIG,
];

const source = "";

describe("Icon", () => {
  it("Renders icon when is SVG correctly", () => {
    const { container } = render(<Icon source={source} />);
    expect(container).toMatchSnapshot();
  });
  it("Renders Icon with other Size number correctly", () => {
    const { container } = render(<Icon source={source} size={16} />);
    expect(container).toMatchSnapshot();
  });
  test.each(ICON_SIZE)("Renders Icon with size %p as argument", (value) => {
    const { container } = render(<Icon source={source} size={value} />);
    expect(container).toMatchSnapshot();
  });
  it("Renders disabled Icon correctly", () => {
    const { container } = render(<Icon source={source} isDisabled />);
    expect(container).toMatchSnapshot();
  });

  it("Renders no disabled Icon correctly", () => {
    const { container } = render(<Icon source={source} isDisabled={false} />);
    expect(container).toMatchSnapshot();
  });
});
