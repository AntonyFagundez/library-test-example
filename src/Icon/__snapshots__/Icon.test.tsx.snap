// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Renders Icon with other Size number correctly 1`] = `
<div>
  <img
    alt="Icono"
    class="root"
    data-size="16"
    src=""
  />
</div>
`;

exports[`Icon Renders Icon with size "BIG" as argument 1`] = `
<div>
  <img
    alt="Icono"
    class="root"
    data-size="BIG"
    src=""
  />
</div>
`;

exports[`Icon Renders Icon with size "LARGE" as argument 1`] = `
<div>
  <img
    alt="Icono"
    class="root"
    data-size="LARGE"
    src=""
  />
</div>
`;

exports[`Icon Renders Icon with size "MEDIUM" as argument 1`] = `
<div>
  <img
    alt="Icono"
    class="root"
    data-size="MEDIUM"
    src=""
  />
</div>
`;

exports[`Icon Renders Icon with size "MSMALL" as argument 1`] = `
<div>
  <img
    alt="Icono"
    class="root"
    data-size="MSMALL"
    src=""
  />
</div>
`;

exports[`Icon Renders Icon with size "REGULAR" as argument 1`] = `
<div>
  <img
    alt="Icono"
    class="root"
    data-size="REGULAR"
    src=""
  />
</div>
`;

exports[`Icon Renders Icon with size "SMALL" as argument 1`] = `
<div>
  <img
    alt="Icono"
    class="root"
    data-size="SMALL"
    src=""
  />
</div>
`;

exports[`Icon Renders Icon with size "XBIG" as argument 1`] = `
<div>
  <img
    alt="Icono"
    class="root"
    data-size="XBIG"
    src=""
  />
</div>
`;

exports[`Icon Renders Icon with size "XMEDIUM" as argument 1`] = `
<div>
  <img
    alt="Icono"
    class="root"
    data-size="XMEDIUM"
    src=""
  />
</div>
`;

exports[`Icon Renders Icon with size "XREGULAR" as argument 1`] = `
<div>
  <img
    alt="Icono"
    class="root"
    data-size="XREGULAR"
    src=""
  />
</div>
`;

exports[`Icon Renders Icon with size "XSMALL" as argument 1`] = `
<div>
  <img
    alt="Icono"
    class="root"
    data-size="XSMALL"
    src=""
  />
</div>
`;

exports[`Icon Renders Icon with size "XXBIG" as argument 1`] = `
<div>
  <img
    alt="Icono"
    class="root"
    data-size="XXBIG"
    src=""
  />
</div>
`;

exports[`Icon Renders Icon with size "XXSMALL" as argument 1`] = `
<div>
  <img
    alt="Icono"
    class="root"
    data-size="XXSMALL"
    src=""
  />
</div>
`;

exports[`Icon Renders disabled Icon correctly 1`] = `
<div>
  <img
    alt="Icono"
    class="root disabled"
    data-size="MEDIUM"
    src=""
  />
</div>
`;

exports[`Icon Renders icon when is SVG correctly 1`] = `
<div>
  <img
    alt="Icono"
    class="root"
    data-size="MEDIUM"
    src=""
  />
</div>
`;

exports[`Icon Renders no disabled Icon correctly 1`] = `
<div>
  <img
    alt="Icono"
    class="root"
    data-size="MEDIUM"
    src=""
  />
</div>
`;
